packages/service-library/src/servicelib/rabbitmq.py

```diff
@@ -137,7 +118,9 @@ async def _get_channel(self) -> aio_pika.abc.AbstractChannel:
         async with self._connection_pool.acquire() as connection:
             connection: aio_pika.RobustConnection
             channel = await connection.channel()
-            channel.close_callbacks.add(_channel_close_callback)
+            channel.close_callbacks.add(
+                functools.partial(channel_close_callback, client=self)
```

Comment:
cannot  `channel_close_callback` be simply a "protected" member function ... this way we can avoid all the issues with "friend" attribute ...

Reply:
yes. I had this before, then I changed I don't why.. but you are right